addon/modifiers/scroll-shadow.ts

```diff
@@ -0,0 +1,61 @@
+// @ts-ignore
+import { setModifierManager } from '@ember/modifier';
+
+type ColorAvailable = 'field' | undefined;
+
+type State = {
+  element: Element;
+  infiniteShadowElement: Element | null;
+  resizeObserver: ResizeObserver;
+  hasScrollbar: boolean;
+};
+
+export default setModifierManager(
+  () => ({
+    createModifier() {
+      return {
+        element: null
+      };
+    },
+
+    baseCSSClass() {
+      return 'scroll-shadow';
+    },
+
+    colorCSSClass(args: any) {
```

Comment:
maybe just pass it the `args.named.color` from outside, and move this to: `colorCSSClass(color: string)` ?